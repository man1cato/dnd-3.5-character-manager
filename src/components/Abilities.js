import React from 'react';

const Abilities = ({ abilities, handleChange }) => (
	<div className="grid grid--abilities">
		<h5 className="grid__col1">Ability</h5>
		{Object.keys(abilities).map((ability, i) =>
			<div className="grid__col1" key={i}>{abilities[ability].name}</div>
		)}

		<h5 className="grid__col2">Score</h5>
		{Object.keys(abilities).map((ability, i) =>
			<div className="grid__col2" key={i}>{abilities[ability].score}</div>
		)}

		<h5 className="grid__col3">Mod</h5>
		{Object.keys(abilities).map((ability, i) =>
			<div className="grid__col3" key={i}>{abilities[ability].mod}</div>
		)}

		<h5 className="grid__col4">Temp Score</h5>
		{Object.keys(abilities).map((ability, i) =>
			<input
			className="grid__col4"
			key={i}
			id={ability}
			name="abilities"
			type="text"
			value={abilities[ability].tempScore}
			onChange={handleChange}
			/>
		)}

		<h5 className="grid__col5">Temp Mod</h5>
		{Object.keys(abilities).map((ability, i) =>
			<div className="grid__col5" key={i}>{abilities[ability].tempMod}</div>
		)}
	</div>
)

export default Abilities;
