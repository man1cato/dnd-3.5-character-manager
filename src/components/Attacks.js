import React from 'react';

const Attacks = (props) => (
    <div>
        <div className="row">Base Attack Bonus: {props.bab}</div>
        <div className=" grid grid--6col">
            <h4 className="grid__left-col">Attack Type</h4>
            <h4>Base</h4>
            <div></div>
            <h4>Temp Mod</h4>
            <div></div>
            <h4>Total</h4>

            <div className="grid__left-col">Melee</div>
            <div>{props.melee.base}</div>
            <div>+</div>
            <input
                type="text"
                name="melee"
                value={props.melee.mod}
                onChange={props.onInputChange}
            />
            <div>=</div>
            <div>{props.melee.total}</div>

            <div className="grid__left-col">Ranged</div>
            <div>{props.ranged.base}</div>
            <div>+</div>
            <input
                type="text"
                name="ranged"
                value={props.ranged.mod}
                onChange={props.onInputChange}
            />
            <div>=</div>
            <div>{props.ranged.total}</div>

            <div className="grid__left-col">Grapple</div>
            <div>{props.grapple.base}</div>
            <div>+</div>
            <input
                type="text"
                name="grapple"
                value={props.grapple.mod}
                onChange={props.onInputChange}
            />
            <div>=</div>
            <div>{props.grapple.total}</div>
        </div>
    </div>
)

export default Attacks