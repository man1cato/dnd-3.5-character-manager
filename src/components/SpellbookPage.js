import React, {Fragment} from 'react';

import { connect } from 'react-redux';
import update from 'immutability-helper';

import Header from './Header';
import Footer from './Footer';
import SpellModal from './SpellModal';
import { startEditProfile } from '../actions/profile';


export class SpellbookPage extends React.Component {
	state = {
		spellbook: this.props.spellbook,
		preparedSpells: [],
		selected: undefined
	}

	handleOpenModal = (e) => {
		const id = e.target.id
		const selected = this.props.spells[id]
		this.setState({selected})
	}

	handleCloseModal = () => {
		this.setState({selected: undefined})
	}

	handleChange = (e) => {
		const spellId = e.target.getAttribute("spellid")
		const level = e.target.getAttribute("level")
		const index = e.target.getAttribute("index")
		const attribute = e.target.getAttribute("attribute")
		const valueChange = Number(e.target.getAttribute("change"))
		const name = e.target.name

		this.setState((prevState) => {
			const spell = prevState.spellbook[level].spells[index]
			const value = prevState.spellbook[level].spells[index][attribute] + valueChange
			const remaining = attribute === "prepared" ? value - spell.used : spell.prepared - value
			let preparedSpells = prevState.preparedSpells;
         if (value > 0 && !preparedSpells.includes(spellId)) {
            preparedSpells.push(spellId)
			}
			if (value === 0 || name === "clear") {
            preparedSpells = preparedSpells.filter((id) => id !== spellId)
			}
			
			if (name === "clear") {
				return {
					spellbook: update(prevState.spellbook, {
						[level]: {
							spells: {
								[index]: {
									prepared: { $set: 0 },
									used: { $set: 0 },
									remaining: { $set: 0 }
								}
							}
						}
					}),
					preparedSpells
				}
			}

			return {
				spellbook: update(prevState.spellbook, {
					[level]: {
						spells: {
							[index]: {
								[attribute]: { $set: value },
								remaining: { $set: remaining }
							}
						}
					}
				}),
				preparedSpells
			}
		}, () => {
			this.setState((prevState) => {
				const total = prevState.spellbook[level].spells.map((spell) => spell.prepared).reduce((total, num) => total + num);
				return {
					spellbook: update(prevState.spellbook, {
						[level]: {
							total: { $set: total }
						}
					})
				}
			}, () => {
				this.props.startEditProfile(this.props.id, {spellbook: this.state.spellbook});
			})
		})
	}
	
	render() {
		return (
			<div className="layout">
				<Header pageTitle="Spellbook" />
				<SpellModal 
					selected={this.state.selected}
					handleCloseModal={this.handleCloseModal}
				/>
				<div className="container container--body">

					{this.state.spellbook.map((page, level) => (
						<div key={level}>
							<h3 className="row row--center">Level {level} Spells ({page.total}/{page.spellsPerDay})</h3>

							<div className="grid grid--spellbook">
								<h5 className="grid__col1">Spell</h5>
								<h5 className="grid--spellbook__school">School</h5>
								<h5>Prep</h5>
								<h5>Rmng</h5>
								<div></div>

								{page.spells.map((spell, i) => (
									<Fragment key={i}>
										<button 
											className="grid__col1 button--link" 
											id={spell.id}
											key={`spell${i}`} 
											onClick={this.handleOpenModal}
										>
											{spell.name}
										</button>

										<div className="grid--spellbook__school" key={`school${i}`}>
											{spell.school.substr(0,4)}
										</div>
										
										<div className="grid__col3 grid--spellbook__attribute" key={`prep${i}`}>
											<button
												change={1}
												spellid={spell.id}
												index={i}
												attribute="prepared"
												level={level}
												onClick={this.handleChange}
											>+</button>
											<div>{spell.prepared}</div>
											<button
												change={-1}
												spellid={spell.id}
												index={i}
												attribute="prepared"
												level={level}
												onClick={this.handleChange}
											>-</button>
										</div>
										
										<div className="grid__col4" key={`rmng${i}`}>{spell.remaining}</div>

										{spell.prepared !== 0 || spell.remaining !== 0 || spell.used !== 0 ?
											<button 
												className="grid__col5" 
												name="clear"
												key={`clear${i}`}
												spellid={spell.id}
												index={i}
												level={level}
												onClick={this.handleChange}
											>
												Clear
											</button>
											:
											<div></div>
										}
									</Fragment>
								))}                        

							</div>
						</div>
					))}

				</div>
				<Footer />
			</div>
		)
	}
}

const mapStateToProps = (state) => ({
	id: state.profile.id,
	spellbook: state.profile.spellbook,
	spells: state.spells
})

const mapDispatchToProps = (dispatch, props) => ({
	startEditProfile: (id, updates) => dispatch(startEditProfile(id, updates))
});


export default connect(mapStateToProps, mapDispatchToProps)(SpellbookPage)