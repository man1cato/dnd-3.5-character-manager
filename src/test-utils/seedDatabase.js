import _ from 'lodash'
import profile from './fixtures/profile'
import database from '../firebase/firebase'


export const characterOne = {
   ..._.omit(profile, ['id'])
}

export const characterTwo = {
   name: 'Korg',
   gender: 'Male',
   race: 'recoDD3gAaQXU3A56',
   jobClass: 'recbxccm8pp3fTmQw',
   alignment: 'Lawful Evil',
   level: 5,
   iconUrl: 'images/orc.svg'
}

export const characterThree = {
   name: 'Rylia',
   gender: 'Female',
   race: 'recFEpQ9zuD9Yqu7l',
   jobClass: 'recQrfCKtksnnSsaZ',
   alignment: 'Chaotic Neutral',
   level: 3,
   iconUrl: 'images/halfling.svg'
}

export const userOne = {
   uid: 'userOne',
   characterIds: ['characterOne', 'characterTwo']
}

export const userTwo = {
   uid: 'userTwo',
   characterIds: ['characterThree']
}


export default async () => {
   await database.ref('users').remove()
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[0]}`).set(characterOne)
   await database.ref(`users/${userOne.uid}/profiles/${userOne.characterIds[1]}`).set(characterTwo)
   await database.ref(`users/${userTwo.uid}/profiles/${userTwo.characterIds[0]}`).set(characterThree)
}