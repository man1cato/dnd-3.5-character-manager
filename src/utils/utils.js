import _ from 'lodash'


export const convertTextToArray = (string) => _.compact(string.split(/\n/))

export const findItemById = (storeItemArray, itemId) => storeItemArray.find((item) => item.id === itemId)

export const apiObjectToArray = (obj) => _.orderBy(Object.keys(obj).map((id) => ({
	id,
	...obj[id]
})), ['name'], ['asc'])