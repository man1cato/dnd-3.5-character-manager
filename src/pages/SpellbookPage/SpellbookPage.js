import React, { Fragment, useState, useEffect } from 'react'
import { connect } from 'react-redux'
import update from 'immutability-helper'
import _ from 'lodash'

import Counter from '../../components/Counter/Counter'
import SpellModal from '../../components/Modals/SpellModal'
import { startEditProfile } from '../../store/actions/profile'
import './SpellbookPage.scss'


const calcPreparedTotals = (spellbook) => _.map(spellbook, (level) => _.reduce(level, (total, spell) => total + spell.prepared, 0))

export const SpellbookPage = (props) => {
	const [spellbook, setSpellbook] = useState(props.profile.spellbook)
	const [preparedTotals, setPreparedTotals] = useState(calcPreparedTotals(spellbook))
	const [clickedSpell, setClickedSpell] = useState(null)
	const spellsPerDay = props.jobClasses[props.profile.jobClass].levels[props.profile.level].spellsPerDay

	useEffect(() => {
		setPreparedTotals(calcPreparedTotals(spellbook))
		props.startEditProfile(props.profile.id, { spellbook })
	}, [spellbook])

	const handleChange = (level, index, value) => {
		const spell = spellbook[level][index]

		setSpellbook(update(spellbook, {
			[level]: {
				[index]: {
					prepared: { $set: value },
					remaining: { $set: value - spell.used }
				}
			}
		}))
	}
	
	return (
		<div className="container--body">
			{spellbook.map((page, level) => (
				<div className="section" key={level}>
					<h3 className="section__title">Level {level} Spells ({preparedTotals[level]}/{spellsPerDay[level]})</h3>

					<div className="spells-grid">
						<h5 className="grid__col1">Spell</h5>
						<h5>Prepared</h5>
						<h5>Rmng</h5>

						{page.map((spell, index) => (
							<Fragment key={index}>
								<button 
									className="grid__col1 button--link" 
									id={spell.id}
									onClick={() => setClickedSpell(props.spells[spell.id])}
								>
									{props.spells[spell.id].name}
								</button>

								<Counter
									value={spell.prepared}
									updateValue={value => handleChange(level, index, value)}
									noInput
								/>
								
								<div id={`${spell.id}Remaining`}>{spell.remaining}</div>

								{(spell.prepared !== 0 || spell.remaining !== 0 || spell.used !== 0) && (
									<button 
										className="clear-button" 
										onClick={() => setSpellbook(update(spellbook, {
											[level]: {
												[index]: {
													prepared: { $set: 0 },
													used: { $set: 0 },
													remaining: { $set: 0 }
												}
											}
										}))}
									>
										<ion-icon name="close-circle" size="small" />
									</button>
								)}
							</Fragment>
						))}                        

					</div>
				</div>
			))}

			<SpellModal
				clickedSpell={clickedSpell}
				handleCloseModal={() => setClickedSpell(undefined)}
			/>

		</div>
	)
	
}

const mapStateToProps = (state) => ({
	profile: state.profile,
	spells: state.spells,
	jobClasses: state.jobClasses
})

const mapDispatchToProps = (dispatch) => ({
	startEditProfile: (id, updates) => dispatch(startEditProfile(id, updates))
})


export default connect(mapStateToProps, mapDispatchToProps)(SpellbookPage)