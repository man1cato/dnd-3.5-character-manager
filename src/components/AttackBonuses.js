import React, { useState, useEffect, Fragment } from 'react'
import _ from 'lodash'
import update from 'immutability-helper'

import { calcBaseMeleeBonus, calcBaseRangedBonus, calcBaseGrappleBonus, convertInputValue } from '../utils/utils'


const AttackBonuses = ({ profileAttackBonusMods, abilities, baseAttackBonuses, size, handleUpdate }) => {  
    const attackBonusBases = {
        melee: _.map(baseAttackBonuses, (bab) => calcBaseMeleeBonus(bab, abilities.str.score, size)),
        ranged: _.map(baseAttackBonuses, (bab) => calcBaseRangedBonus(bab, abilities.dex.score, size)),
        grapple: _.map(baseAttackBonuses, (bab) => calcBaseGrappleBonus(bab, abilities.str.score, size))
    }
    const [attackBonusMods, setAttackBonusMods] = useState(_.mapValues(profileAttackBonusMods, (val) => val || '')) 
    const [attackBonusTotals, setAttackBonusTotals] = useState(
        _.mapValues(attackBonusMods, (mod, key) => _.map(attackBonusBases[key], (val) => Number(mod) + val ))
    ) 
    
    useEffect(() => {
        setAttackBonusTotals(_.mapValues(attackBonusMods, (mod, key) => _.map(attackBonusBases[key], (val) => Number(mod) + val)))
        handleUpdate({ attackBonusMods })
    }, [attackBonusMods])

    return (    
        <div className="grid--6col">
            <h5 className="grid__col1">Type</h5>
            <h5>Base</h5>
            <div></div>
            <h5>Mod</h5>
            <div></div>
            <h5>Total</h5>

            {_.map(['melee', 'ranged', 'grapple'], (key) => (
                <Fragment key={key}>
                    <div className="grid__col1">{_.startCase(key)}</div>
                    <div>{attackBonusBases[key].join(' / ')}</div>
                    <div>+</div>
                    <input
                        type="number"
                        id={key}
                        value={attackBonusMods[key]}
                        onChange={(e) => {
                            setAttackBonusMods(
                                update(attackBonusMods, {
                                    [key]: { $set: convertInputValue(e.target.value) }
                                })
                            )
                        }}
                    />
                    <div>=</div>
                    <div>{attackBonusTotals[key].join(' / ')}</div>
                </Fragment>
            ))}

        </div>
    )
}


export default AttackBonuses