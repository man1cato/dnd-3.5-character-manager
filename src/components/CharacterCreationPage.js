import React from 'react'
import {connect} from 'react-redux'
import Header from './Header'
import {Formik, Form} from 'formik'
import * as Yup from 'yup'

import Page1 from './CreatorFormPage1'
import Page2 from './CreatorFormPage2'
import Page3 from './CreatorFormPage3'
import CreatorFormFooter from './CreatorFormFooter'
import {history} from '../routers/AppRouter'
import {startCreateProfile} from '../actions/profile'
import {getSpecialAbilityIdsFromLevels} from '../utils/getSpecialAbilities'


const pages = [Page1, Page2, Page3]

const validationSchema = Yup.object().shape({
	page1: Yup.object().shape({
		name: Yup.string().min(2, 'Name is too short!').max(30, 'Name is too long!').required('Required'),
		age: Yup.number().required().positive().integer(),
		heightFt: Yup.number('Enter a positive integer').positive('Enter a positive integer').integer('Enter a positive integer').required('Required'),
		heightIn: Yup.number('Enter an integer between 0 and 12').positive('Enter an integer between 0 and 12').integer('Enter an integer between 0 and 12').max(11, 'Enter an integer between 0 and 12').required('Required'),
		weight: Yup.number('Enter a positive integer').positive('Enter a positive integer').integer('Enter a positive integer').required('Required')
	}),
	page2: undefined,
	page3: undefined
})

export class CharacterCreationPage extends React.Component {
	state = {
		selectedRace: this.props.races.find((race) => race.name === 'Human'),
		selectedJobClass: this.props.jobClasses.find((jobClass) => jobClass.name === 'Fighter'),
		page: 1
	}

	handleSelect = (e, setFieldValue, setFieldError) => {
		const name = e.target.name;
		const value = e.target.value;
		if (name === 'race') { setFieldValue('bonusLanguages', []) } 
		if (name === 'school') {
			setFieldValue('prohibitedSchools', []);
			if (value === 'Universal') {
				setFieldError('prohibitedSchools', undefined);
			} 
		} 
		if (name === 'jobClass') {
			if (value === this.props.jobClasses.find((jobClass) => jobClass.name === 'Paladin').id) { 
				setFieldValue('alignment', 'Lawful Good') 
			}
			if (value === this.props.jobClasses.find((jobClass) => jobClass.name === 'Wizard').id) { 
				setFieldValue('school', 'Universal') 
			} else {
				setFieldValue('school', null)
			}
		}
		
		this.setState((prevState) => {
			const selectedRace = name === 'race' ? this.props.races.find((race) => race.id === value) : prevState.selectedRace;
			const selectedJobClass = name === 'jobClass' ? this.props.jobClasses.find((jobClass) => jobClass.id === value) : prevState.selectedJobClass;
			
			return {
				selectedRace,
				selectedJobClass
			}
		});
	}

	handleMultiSelect = (e, setFieldValue) => {
		const name = e.target.name;
		const options = e.target.options;
		const value = [];
		for (let i = 0; i < options.length; i++) {
			if (options[i].selected) {
				value.push(options[i].value);
			}
		}
		setFieldValue(name, value);
	}

	handleBack = (setErrors) => {
		this.setState((prevState) => ({
			page: prevState.page - 1
		}));        
		setErrors({});
	}

	handleNext = () => {
		this.setState((prevState) => ({
			page: prevState.page + 1
		}));
	}    

	render() {        
		return (
			<div className="layout">
				<Header pageTitle="Character Creation" />
				
				<Formik
					initialValues={{
						name: '',
						gender: 'Male',
						age: '',
						heightFt: '',
						heightIn: '',
						weight: '',
						race: this.props.races.find((race) => race.name === 'Human').id,
						jobClass: this.props.jobClasses.find((jobClass) => jobClass.name === 'Fighter').id,
						alignment: 'Lawful Good',
						deity: ''
					}}
												
					validationSchema={Yup.reach(validationSchema, `page${this.state.page}`)}

					onSubmit={(values, {setErrors, setSubmitting}) => {
						getSpecialAbilityIdsFromLevels(values.jobClass, 1).then((specialAbilities) => {
							const profile = {
								...values,
								race: this.state.selectedRace.name,
								size: this.state.selectedRace.size,
								specialAbilities,
								jobClass: this.state.selectedJobClass.name,
								height: `${values.heightFt}'${values.heightIn}"`,
								languages: this.state.selectedRace.defaultLanguages.concat(values.bonusLanguages),
								deity: !!values.deity ? values.deity : "None",
								level: 1,
								iconUrl: this.state.selectedRace.iconUrl
							}
							delete profile.heightFt
							delete profile.heightIn
							if(profile.school === null) {delete profile.school}
							this.props.startCreateProfile(profile)
						})							
						setTimeout(() => { history.push('/profile') }, 2000)
						setSubmitting(false)
					}}
				>

					{({values, setFieldValue, handleChange, isSubmitting, isValid, setErrors, setFieldError}) => (
						<Form >
							<div className="container container--body">
								{{
									1: <Page1 
										values={values}
										races={this.props.races}
										jobClasses={this.props.jobClasses}
										selectedRace={this.state.selectedRace} 
										handleChange={handleChange} 
										handleSelect={this.handleSelect}  
										setFieldValue={setFieldValue}
									/>,
									2: <Page2
										values={values}
										selectedRace={this.state.selectedRace} 
										jobClasses={this.props.jobClasses}
										selectedJobClass={this.state.selectedJobClass}
										alignments={this.props.alignments}
										schools={this.props.schools}
										handleChange={handleChange}
										handleSelect={this.handleSelect}
										handleMultiSelect={this.handleMultiSelect}
										setFieldValue={setFieldValue}
										setFieldError={setFieldError}
									/>,
									3: <Page3
										values={values}
										feats={this.props.feats}
										selectedJobClass={this.state.selectedJobClass}										
										handleChange={handleChange}
										handleSelect={this.handleSelect}
										handleMultiSelect={this.handleMultiSelect}
										setFieldValue={setFieldValue}
										setFieldError={setFieldError}
									/>
								}[this.state.page]}
							</div>
							
							<CreatorFormFooter 
								page={this.state.page}
								pages={pages}
								handleBack={this.handleBack}
								handleNext={this.handleNext}
								handleSubmit={this.handleSubmit}
								setErrors={setErrors}
								isSubmitting={isSubmitting}
								isValid={isValid}
							/>

						</Form>
					)}
				</Formik>
					
			</div>
		)
	}
};


const mapStateToProps = (state) => ({
	races: state.races,
	jobClasses: state.jobClasses,
	feats: state.feats,
	schools: [
		'Abjuration','Clairsentience', 'Conjuration', 'Divination', 
		'Enchantment', 'Evocation', 'Illusion', 'Metacreativity', 
		'Necromancy', 'Psychokinesis', 'Psychometabolism', 'Psychoportation', 
		'Telepathy', 'Transmutation', 'Universal'
	],
	alignments: [
		'Lawful Good', 'Neutral Good', 'Chaotic Good', 
		'Lawful Neutral', 'True Neutral', 'Chaotic Neutral', 
		'Lawful Evil', 'Neutral Evil', 'Chaotic Evil'
	]
});

const mapDispatchToProps = (dispatch, props) => ({
	startCreateProfile: (profile) => dispatch(startCreateProfile(profile))
});


export default connect(mapStateToProps, mapDispatchToProps)(CharacterCreationPage);
