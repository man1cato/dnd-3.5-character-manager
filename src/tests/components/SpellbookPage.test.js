import React from 'react'
import { shallow } from 'enzyme'

import { SpellbookPage } from '../../components/SpellbookPage'
import { characterOne } from '../utils/seedDatabase'
import { apiData } from '../utils/utils'


let props, wrapper
beforeAll(async () => {
	const api = await apiData()
    props = {
        id: 'characterOne',
        spellbook: characterOne.spellbook,
        spells: api.spells
    }
    wrapper = shallow(<SpellbookPage {...props} />)
})

test('should render SpellbookPage with profile data', () => {
    expect(wrapper).toMatchSnapshot()
})