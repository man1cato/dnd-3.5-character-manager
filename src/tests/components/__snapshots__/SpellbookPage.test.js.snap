// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SpellbookPage with profile data 1`] = `
<div
  className="layout"
>
  <Connect(Header)
    pageTitle="Spellbook"
  />
  <SpellModal
    handleCloseModal={[Function]}
  />
  <div
    className="container container--body"
  >
    <div
      key="0"
    >
      <h3
        className="row--center"
      >
        Level 
        0
         Spells (
        /
        4
        )
      </h3>
      <div
        className="grid grid--spellbook"
      >
        <h5
          className="grid__col1"
        >
          Spell
        </h5>
        <h5
          className="grid--spellbook__school"
        >
          School
        </h5>
        <h5>
          Prep
        </h5>
        <h5>
          Rmng
        </h5>
        <div />
        <button
          className="grid__col1 button--link"
          id="rec8IQeYmBr3f1BMC"
          onClick={[Function]}
        >
          Acid Splash
        </button>
        <div
          className="grid--spellbook__school"
        >
          Conj
        </div>
        <div
          className="grid__col3 grid--spellbook__attribute"
        >
          <button
            attribute="prepared"
            change={1}
            index={0}
            level={0}
            onClick={[Function]}
            spellid="rec8IQeYmBr3f1BMC"
          >
            +
          </button>
          <div>
            1
          </div>
          <button
            attribute="prepared"
            change={-1}
            index={0}
            level={0}
            onClick={[Function]}
            spellid="rec8IQeYmBr3f1BMC"
          >
            -
          </button>
        </div>
        <div
          className="grid__col4"
        >
          1
        </div>
        <button
          className="grid__col5"
          index={0}
          level={0}
          name="clear"
          onClick={[Function]}
          spellid="rec8IQeYmBr3f1BMC"
        >
          Clear
        </button>
        <button
          className="grid__col1 button--link"
          id="recwpeRoCsea4vidW"
          onClick={[Function]}
        >
          Arcane Mark
        </button>
        <div
          className="grid--spellbook__school"
        >
          Univ
        </div>
        <div
          className="grid__col3 grid--spellbook__attribute"
        >
          <button
            attribute="prepared"
            change={1}
            index={1}
            level={0}
            onClick={[Function]}
            spellid="recwpeRoCsea4vidW"
          >
            +
          </button>
          <div>
            2
          </div>
          <button
            attribute="prepared"
            change={-1}
            index={1}
            level={0}
            onClick={[Function]}
            spellid="recwpeRoCsea4vidW"
          >
            -
          </button>
        </div>
        <div
          className="grid__col4"
        >
          1
        </div>
        <button
          className="grid__col5"
          index={1}
          level={0}
          name="clear"
          onClick={[Function]}
          spellid="recwpeRoCsea4vidW"
        >
          Clear
        </button>
      </div>
    </div>
    <div
      key="1"
    >
      <h3
        className="row--center"
      >
        Level 
        1
         Spells (
        /
        3
        )
      </h3>
      <div
        className="grid grid--spellbook"
      >
        <h5
          className="grid__col1"
        >
          Spell
        </h5>
        <h5
          className="grid--spellbook__school"
        >
          School
        </h5>
        <h5>
          Prep
        </h5>
        <h5>
          Rmng
        </h5>
        <div />
        <button
          className="grid__col1 button--link"
          id="recRhVs0ahtQZpbXx"
          onClick={[Function]}
        >
          Charm Person
        </button>
        <div
          className="grid--spellbook__school"
        >
          Ench
        </div>
        <div
          className="grid__col3 grid--spellbook__attribute"
        >
          <button
            attribute="prepared"
            change={1}
            index={0}
            level={1}
            onClick={[Function]}
            spellid="recRhVs0ahtQZpbXx"
          >
            +
          </button>
          <div>
            1
          </div>
          <button
            attribute="prepared"
            change={-1}
            index={0}
            level={1}
            onClick={[Function]}
            spellid="recRhVs0ahtQZpbXx"
          >
            -
          </button>
        </div>
        <div
          className="grid__col4"
        >
          0
        </div>
        <button
          className="grid__col5"
          index={0}
          level={1}
          name="clear"
          onClick={[Function]}
          spellid="recRhVs0ahtQZpbXx"
        >
          Clear
        </button>
      </div>
    </div>
  </div>
  <Connect(Footer) />
</div>
`;
